### PR TITLE
add python sample, update templates and bump rust deps

### PR DESCRIPTION
**Pull request by karthik2804**
Add an example in python and also update dependencies of rust example/template.

### CODE REVIEW

**Review by fibonacci1729 — APPROVED** [review]
I would still make the `wit-bindgen` dep a workspace dependency and maybe bump.

File: guest-py/README.md
Comment (itowlson) [inline guest-py/README.md:1]: Any specific reason for `py` rather than `python`?

**Review by itowlson — APPROVED** [review]
Nice!  It would be good to change the current `guest` directory to `guest-rust` but not a blocker.

File: guest-py/README.md
Comment (karthik2804) [reply guest-py/README.md:1]: updated.

**Pull request merged**